src/scroll-timeline-css-parser.js

```diff
@@ -135,16 +135,14 @@ export class StyleParser {
     return null;
   }
 
-  getViewTimelineOptions(timelineName) {
-    // TODO: Take into account the scoping of the ViewTimelines
-    // https://github.com/w3c/csswg-drafts/issues/7047
+  getViewTimelineOptions(timelineName, target) {
     for (let i = this.subjectSelectorToViewTimeline.length - 1; i >= 0; i--) {
       const options = this.subjectSelectorToViewTimeline[i];
       if(options.name == timelineName) {
-        const allSubjects = document.querySelectorAll(options.selector);
-        if(allSubjects.length) {
+        const subject = target.closest(options.selector);
```

Comment:
https://drafts.csswg.org/scroll-animations-1/#timeline-scope

My reading of "closest" is that it is an inclusive ancestor and does not consider upstream slblings.  Let me take a quick read of the Blink implementation to see how we dealt with this.

Reply:
Fixed in c98c9e2e98f832376de77204f2cd8df8519a0901

Standalone version of that function to (manually) test things: https://codepen.io/bramus/pen/RwyJOJM/1f63d147250c7f682be221b71f0e1961